### PR TITLE
EZP-29275 As a QA I want to cover scenarios of testing fields editing in content management

### PR DESCRIPTION
**Pull request by m-tyrala**
| Question      | Answer
| ------------- | ---
| Tickets       | 
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | no
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

As a QA
I want to cover with tests ez-fields (all usable except of media, binary and image) within the area of basic management of content edition
To increase the detection of errors in this area

#### Checklist:
- [ ] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review

### CODE REVIEW

File: features/ContentTypeFields.feature [hunk 1]
```diff
@@ -20,26 +20,77 @@ Feature: Content fields setting and editing
...
+      | ezauthor             | Authors                      | name      | Test Name             | email      | email@example.com     |         |          | Test Name                 |
+      | ezboolean            | Checkbox                     | value     | true                  |            |                       |         |          | 1                         |
+      | ezobjectrelation     | Content relation (single)    | value     | Media/Images          |            |                       |         |          | Images                    |
+      | ezobjectrelationlist | Content relations (multiple) | firstItem | Media/Images          | secondItem | Media/Files           |         |          | Images Files              |
+      | ezcountry            | Country                      | value     | Poland                |            |                       |         |          | Poland                    |
+      | ezdate               | Date                         | value     | 11/23/2019            |            |                       |         |          | Saturday 23 November 2019 |
```
[the remark this answers is not in the document]
Comment (m-tyrala) [reply features/ContentTypeFields.feature]: Of course - our implementation of date picker does not allow us to select a date before the date currently selected. At first, I was trying to set earlier date in edit scenario. When I realized that, I just swapped the dates places.

File: src/lib/Behat/BusinessContext/ContentUpdateContext.php [hunk 2]
```diff
@@ -20,7 +20,10 @@ public function iSetFields(TableNode $table): void
         $updateItemPage = PageObjectFactory::createPage($this->utilityContext, ContentUpdateItemPage::PAGE_NAME, '');
         $hash = $table->getHash();
         foreach ($hash as $row) {
-            $updateItemPage->contentUpdateForm->fillFieldWithValue($row['label'], $row);
+            $values = $row;
+            unset($values['label']);
+            $values2 = array_filter($values, function ($element) { return !empty($element) || $element === 0;});
```
[the remark this answers is not in the document]
Comment (mnocon) [reply src/lib/Behat/BusinessContext/ContentUpdateContext.php]: 👍 For clarity I'd extract it to a function, something like `filterOutNonEmptyParameters`. BTW, naming filters seems hard 😄 (https://stackoverflow.com/questions/45939202/filter-naming-convention)

**Comment by m-tyrala** [comment]
Just one known problem with notification, not related to this PR.

File: src/lib/Behat/PageObject/ContentItemPage.php [hunk 3]
```diff
@@ -92,7 +92,7 @@ public function goToSubItem(string $contentName, string $contentType): void
     public function navigateToPath(string $path): void
     {
         $pathArray = explode('/', $path);
-        $menuTab = ($pathArray[0] === 'Home') ? 'Content structure' : $pathArray[0];
+        $menuTab = ($pathArray[0] === 'Home' || $pathArray[0] === 'eZ Platform') ? 'Content structure' : $pathArray[0];
```
Comment (mnocon) [inline src/lib/Behat/PageObject/ContentItemPage.php]: You can use EzEnvConst here and remove the OR.

**Comment by lserwatka** [comment]
You have 16 commits here. Could you squash it?

**Comment by m-tyrala** [comment]
@lserwatka finally, this one can be merged

**Pull request merged**